example/src/main/java/org/eclipse/hono/example/AbstractExampleClient.java

```diff
@@ -57,15 +61,34 @@ public final void setActiveProfiles(final Environment env) {
     @Autowired
     public final void setVertx(final Vertx vertx) {
         this.vertx = vertx;
-        this.ctx = vertx.getOrCreateContext();
+        ctx = vertx.getOrCreateContext();
     }
 
     @Autowired
     public final void setHonoClient(final HonoClient client) {
         this.client = Objects.requireNonNull(client);
     }
 
+    @Autowired
+    public final void setClientConfigProperties(final ClientConfigProperties config) {
+        this.clientConfigProperties = Objects.requireNonNull(config);
+    }
+
     protected final ProtonClientOptions getClientOptions() {
-        return new ProtonClientOptions().setConnectTimeout(DEFAULT_CONNECT_TIMEOUT_MILLIS).setReconnectAttempts(2);
+        ProtonClientOptions protonClientOptions = new ProtonClientOptions()
+                .setConnectTimeout(DEFAULT_CONNECT_TIMEOUT_MILLIS).setReconnectAttempts(2);
+        if (!StringUtils.isEmpty(clientConfigProperties.getProxyHost()) &&
```

Comment:
all of the following code should be moved to `ClientConfigProperties`, e.g. to a `getProxyOptions` method, so that it can be used by other components as well